src/envoy/http/authn/origin_authenticator_test.cc

```diff
@@ -219,6 +303,54 @@ TEST_P(OriginAuthenticatorTest, MultipleFail) {
                                       filter_context_.authenticationResult()));
 }
 
+TEST_P(OriginAuthenticatorTest, MultipleRuleTriggeredValidationSucceeded) {
+  ASSERT_TRUE(Protobuf::TextFormat::ParseFromString(
+      kMultipleOriginMethodWithTriggerRulePolicy, &policy_));
+
+  createAuthenticator();
+  // First method triggered but failed, second method not triggered, third
+  // method triggered and succeeded.
+  EXPECT_CALL(*authenticator_, validateJwt(_, _))
+      .Times(2)
+      .WillOnce(DoAll(SetArgPointee<1>(jwt_extra_payload_), Return(false)))
+      .WillOnce(DoAll(SetArgPointee<1>(jwt_payload_), Return(true)));
+
+  setPath("/allow");
+  EXPECT_TRUE(authenticator_->run(payload_));
+  EXPECT_TRUE(TestUtility::protoEqual(expected_result_when_pass_,
+                                      filter_context_.authenticationResult()));
+}
+
+TEST_P(OriginAuthenticatorTest, MultipleRuleTriggeredValidationFailed) {
+  ASSERT_TRUE(Protobuf::TextFormat::ParseFromString(
+      kMultipleOriginMethodWithTriggerRulePolicy, &policy_));
+
+  createAuthenticator();
+  // Triggered on first and second method but all failed.
+  EXPECT_CALL(*authenticator_, validateJwt(_, _))
+      .Times(2)
+      .WillRepeatedly(
+          DoAll(SetArgPointee<1>(jwt_extra_payload_), Return(false)));
+
+  setPath("/two");
+  EXPECT_FALSE(authenticator_->run(payload_));
+  EXPECT_TRUE(TestUtility::protoEqual(initial_result_,
+                                      filter_context_.authenticationResult()));
+}
+
+TEST_P(OriginAuthenticatorTest, MultipleRuleNotTriggered) {
+  ASSERT_TRUE(Protobuf::TextFormat::ParseFromString(
+      kMultipleOriginMethodWithTriggerRulePolicy, &policy_));
+
+  createAuthenticator();
+  EXPECT_CALL(*authenticator_, validateJwt(_, _)).Times(0);
+
+  setPath("/bad");
+  EXPECT_TRUE(authenticator_->run(payload_));
+  EXPECT_TRUE(TestUtility::protoEqual(initial_result_,
+                                      filter_context_.authenticationResult()));
+}
+
```

Comment:
Add a test for the case that path is not existed in headers?

Reply:
Done